### ISSUE TITLE
Psychopy with sounddevice backend can't play sounds that sounddevice can

### ISSUE THREAD

**Issue by mwaskom** [psychopy/psychopy#1820]
I have some experiment code that plays some sounds. I have been using the pygame backend, but am aiming to move over to sounddevice.

I've found that some of my .wav files could be played by PsychoPy with the pygame backend but cannot be played by PsychoPy with the sounddevice backend. Surprisingly, I have no problem paying these files directly using soundfile and sounddevice.

Here's an example small wav file that I have trouble with: [secalert.zip](https://github.com/psychopy/psychopy/files/1985852/secalert.wav.zip)

(I compressed it so that github will let me upload it; the underlying file is only 12Kb).

This won't play using a simple PsychoPy example

```python
from psychopy import sound
sound.Sound("secalert.wav").play()
```

Sometimes an exception makes its way to stderr (but is not raised in the Python interpreter):

```python-traceback
Traceback (most recent call last):
  File "/Users/mwaskom/anaconda/envs/psychopy/lib/python2.7/site-packages/sounddevice.py", line 733, in callback_ptr
    return _wrap_callback(callback, data, frames, time, status)
  File "/Users/mwaskom/anaconda/envs/psychopy/lib/python2.7/site-packages/sounddevice.py", line 2508, in _wrap_callback
    callback(*args)
  File "/Users/mwaskom/anaconda/envs/psychopy/lib/python2.7/site-packages/psychopy/sound/backend_sounddevice.py", line 186, in callback
    dat = thisSound._nextBlock()  # fetch the next block of data
  File "/Users/mwaskom/anaconda/envs/psychopy/lib/python2.7/site-packages/psychopy/sound/backend_sounddevice.py", line 489, in _nextBlock
    .format(self.stereo))
```

But I have no problem playing the file using the underlying libraries without any bells or even whistles:

```python
import soundfile
import sounddevice
array, bitrate = soundfile.read("secalert.wav")
sounddevice.play(array, bitrate)
```

On the small sample of files I have used in my experiments, it seems like stereo files work fine but mono ones break.

If I force loading in mono (`sound.Sound("secalert.wav", stereo=False)`), then I get a distorted version of the sound, and *all other* sounds from my computer (including from other apps) are distorted until I exit the Python session.

Versions used:
- psychopy: 1.90.1 (also present on older versions)
- sounddevice: 0.3.10
- soundfile: 0.9.0

**Comment by mwaskom** [comment]
I'm on MacOS Sierra. How do you check the PortAudio version?

**Comment by CRiddler** [comment]
import sounddevice
    sounddevice.get_portaudio_version()

**Comment by mwaskom** [comment]
Thanks; that shows

```
(1246720,
 u'PortAudio V19.6.0-devel, revision 396fe4b6699ae929d3a685b3ef8a7e97396139a4')
```

**Comment by mwaskom** [comment]
FWIW I'm now also having trouble replicating the distorted audio with the simple example; have run some OS updates in the meantime, maybe that fixed something...

**Comment by mwaskom** [comment]
Though I guess now what's happening is that, if I've tried to load the .wav file without specifying stereo previously, and then load it again (same session) setting `stereo=False`, I don't hear anything when I call `.play()`.

**Comment by CRiddler** [comment]
Ahh, well thats almost good news because I couldn't replicate the sound distortion at all with sound file you provided. If you have another audio file/parameters that can cause the distorted audio I'll take another look into that.

As to the more recent issue, are you running all of this from the interpreter? Or from a script? 

From the interpreter, this plays both sounds for me

    >>> from psychopy import sound
    >>> import os.path
    >>> sfile = os.path.expanduser('~/Desktop/secalert.wav')
    >>> sound.Sound(sfile).play()
    >>> sound.Sound(sfile, stereo=False).play()


If you're running from a script like so:

    #foo.py
    from psychopy import sound
    import os.path

    sfile = os.path.expanduser('~/Desktop/secalert.wav')
    sound.Sound(sfile).play()
    sound.Sound(sfile, stereo=False).play()
`python foo.py`

This doesn't play any sound for me, because the script is ending before the sounds are actually able to be played. Using time.sleep(1) fixes this issue for me:

    #foo.py
    from psychopy import sound
    import os.path
    import time

    sfile = os.path.expanduser('~/Desktop/secalert.wav')
    sound.Sound(sfile).play()
    time.sleep(1)
    sound.Sound(sfile, stereo=False).play()
    time.sleep(1)
`python foo.py`

Let me know if I missed what you were describing in your most recent comment.

**Comment by mwaskom** [comment]
For testing the simple examples in the issue I was running from the interpreter but I ran into the problem originally in a script context.

> From the interpreter, this plays both sounds for me

Is this after you fixed the issue you identified upthread?

**Comment by CRiddler** [comment]
Oh yes it is. I just moved those 8 or so lines out one indentation level. I’ll put that in a PR tomorrow when I have some time.

**Comment by peircej** [comment]
This is included in the new release (1.90.2)
